typescript/object.go

```diff
@@ -0,0 +1,37 @@
+package typescript
+
+// ObjectValue represents a javascript object
+type ObjectValue struct {
+	BlockDeclaration
+}
+
+// WriteCode implements Writable
+func (t *ObjectValue) WriteCode(writer CodeWriter) {
```

Comment:
this code here has been echoed in a few places. If someone thins the repetition is no bueno, I can try to dry it out.